docs/distribution/launcher-cli.md

```diff
@@ -0,0 +1,154 @@
+---
+layout: developer-doc
+title: Enso Launcher CLI
+category: distribution
+tags: [distribution, launcher, cli, command]
+order: 4
+---
+
+# Enso Launcher CLI
+This document describes available command-line options of the Enso launcher.
+
+This document is a DRAFT. The explanations are just to give an idea of the
+commands. It will be updated with detailed explanations when the CLI is
+developed.
+
+<!-- MarkdownTOC levels="2,3" autolink="true" -->
+
+- [Commands](#commands)
+  - [`new`](#new)
+  - [`install`](#install)
+  - [`uninstall`](#uninstall)
+  - [`list`](#list)
+  - [`default`](#default)
+  - [`config`](#config)
+  - [`run`](#run)
+  - [`repl`](#repl)
+  - [`language-server`](#language-server)
+  - [`upgrade`](#upgrade)
+  - [`version`](#version)
+  - [`help`](#help)
+- [General Options](#general-options)
+  - [`--path`](#--path)
+  - [`--version`](#--version)
+  - [`--use-system-jvm`](#--use-system-jvm)
+
+<!-- /MarkdownTOC -->
+
+## Commands
+
+### `new`
+Create a new, empty project in a specified directory.
+By default uses the `default` enso version, which can be overriden with
+`--version`.
+
+Examples:
+```bash
+enso new project1
+enso new project2 --version 2.0
+```
+
+### `install`
+Installs a specific version of Enso.
+
+Examples:
+```bash
+enso install 2.0.1
+```
+
+### `uninstall`
+Uninstalls a specific version of Enso.
+
+Examples:
+```bash
+enso uninstall 2.0.1
+```
+
+### `list`
+Lists all installed versions of Enso and managed GraalVM distributions.
```

Comment:
My initial idea was to relate Enso and Graal versions, for example like this:

```
> enso list
Enso 2.0.1 -> GraalVM 20.1.0-java11
Enso 2.2.3 -> GraalVM 20.1.0-java11
Enso 3.0.0 -> GraalVM 21.1.0-java14
```

Otherwise, I understand we can display enso and graal versions separately. Should it at least include how many enso installations use a given graal version? Like this:

```
> enso list enso
2.0.1
2.2.3
3.0.0
> enso list graalvm
GraalVM 20.1.0-java11 (used by 2 Enso installations)
GraalVM 21.1.0-java14 (used by 1 Enso installation)
```

Reply:
Oooh, both of those are fantastic. I'd make the last one `enso list runtime` perhaps?

Reply:
So maybe the default `list` without arguments should show this first result, and `enso list runtime` would list Graal versions but without the summary (how many installations use it), since it is apparent from `enso list`?

Reply:
My suggestion was that `enso list` should behave as exactly as you've shown, and then `enso list enso` and `enso list runtime` should behave as you've shown below it.